mlflow/lightgbm.py

```diff
@@ -462,13 +471,22 @@ def log_feature_importance_plot(features, importance, importance_type):
             )
             if early_stopping:
                 extra_step = len(eval_results)
-
-                metrics_logger.record_metrics({"stopped_iteration": extra_step})
-                # best_iteration is set even if training does not stop early.
-                metrics_logger.record_metrics({"best_iteration": model.best_iteration})
+                autologging_client.log_metrics(
+                    run_id=mlflow.active_run().info.run_id,
+                    metrics={
+                        "stopped_iteration": extra_step,
+                        # best_iteration is set even if training does not stop early.
+                        "best_iteration": model.best_iteration,
+                    },
+                )
                 # iteration starts from 1 in LightGBM.
-                results = eval_results[model.best_iteration - 1]
-                metrics_logger.record_metrics(results, step=extra_step)
```

Comment:
`BatchMetricsLogger` is tailored to per-epoch metric logging use cases from callbacks; it may make multiple API calls during this segment. The new `MlflowAutologgingQueueingClient` is a better choice here; it will only make a single API call to LogBatch.

Reply:
Nice.